account_avatax_oca/hooks.py

```diff
@@ -13,3 +18,85 @@ def pre_init_hook(cr):
+                handle_price_include = True
+                quantity = line.quantity
+            else:
+                amount_currency = line.amount_currency
+                handle_price_include = False
+                quantity = 1
+            compute_all_currency = line.tax_ids.with_context(
+                current_aml=line.id
+            ).compute_all(
+                amount_currency,
+                currency=line.currency_id,
+                quantity=quantity,
+                product=line.product_id,
+                partner=line.move_id.partner_id or line.partner_id,
+                is_refund=line.is_refund,
+                handle_price_include=handle_price_include,
+                include_caba_tags=line.move_id.always_tax_exigible,
+                fixed_multiplicator=sign,
+            )
+            rate = line.amount_currency / line.balance if line.balance else 1
+            line.compute_all_tax_dirty = True
+            line.compute_all_tax = {
+                frozendict(
+                    {
+                        "tax_repartition_line_id": tax["tax_repartition_line_id"],
+                        "group_tax_id": tax["group"] and tax["group"].id or False,
+                        "account_id": tax["account_id"] or line.account_id.id,
+                        "currency_id": line.currency_id.id,
+                        "analytic_distribution": (
+                            (tax["analytic"] or not tax["use_in_tax_closing"])
+                            and line.move_id.state == "draft"
+                        )
+                        and line.analytic_distribution,
+                        "tax_ids": [(6, 0, tax["tax_ids"])],
+                        "tax_tag_ids": [(6, 0, tax["tag_ids"])],
+                        "partner_id": line.move_id.partner_id.id or line.partner_id.id,
+                        "move_id": line.move_id.id,
+                        "display_type": line.display_type,
+                    }
+                ): {
+                    "name": tax["name"]
+                    + (" " + _("(Discount)") if line.display_type == "epd" else ""),
+                    "balance": tax["amount"] / rate,
+                    "amount_currency": tax["amount"],
+                    "tax_base_amount": tax["base"]
+                    / rate
+                    * (-1 if line.tax_tag_invert else 1),
+                }
+                for tax in compute_all_currency["taxes"]
+                if tax["amount"]
+            }
+            if not line.tax_repartition_line_id:
+                line.compute_all_tax[frozendict({"id": line.id})] = {
+                    "tax_tag_ids": [(6, 0, compute_all_currency["base_tags"])],
+                }
+
+    if not hasattr(AccountMoveLine, "_compute_all_tax_origin"):
+        AccountMoveLine._compute_all_tax_origin = AccountMoveLine._compute_all_tax
+    AccountMoveLine._compute_all_tax = _compute_all_tax_new
```

Comment:
It looks like we are monkey patching the _compute_all_tax method.
Why would we need to do that?

Reply:
It's needed because in this line https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move_line.py#L940, I'm adding a context with current aml, to know what account move line trigger tax compute, then in next process I can process with more precision